test/check-networking

```diff
@@ -45,6 +45,7 @@ class TestNetworking(MachineCase):
             element = self.browser.text("#networking-interfaces tr:contains('%s')" % iface)
             for state in valid_states:
                 if state in element:
+                    self.browser.wait_visible("#networking-interfaces tr:contains('%s')" % iface)
                     return
             time.sleep(0.5)
```

Comment:
(I started writing this as a quick comment, but I guess this should go to the documentation somewhere, after checking for bugs.)

While we sleep here in the test case, phantomjs is blocked and waits for the next command.  Nothing happens in the browser, it doesn't process external events, etc.  It only runs its event loop during certain commands, such as `wait`, `open`, `expect_reload`, etc.

If you want to wait for a more complex condition that what is provided by the Browser class (such as `wait_in_text`, `wait_visible`, etc.) you can wait for a arbitrary JavaScript expression to become true.  For example:

```
b.wait_js_cond('ph_in_text("#bla", "foo") || ph_in_text("bla", "bar")')
```

If you need to pass run-time values to JavaScript, it is best to pass them as arguments to a anonymous function.  Testlib.py takes care of translating the values from Python to JavaScript (via JSON):

```
b.wait_js_func('(function (sel, val1, val2) { return ph_in_text(sel, val1) || ph_in_text(sel, val2); })',
                        "#bla", "foo", "bar")
```

(There is some parsing weirdness as one could expect from JavaScript, so it's best to wrap the function in parens, like above, and to not start or end the string with newlines.)

If you don't want to mix JavaScript and Python too much (which is a good idea), and want to write a explicit loop in Python that checks the condition with Python functions like `is_visible` or things like `"foo" in b.text("#bla")`, you can use `waitcheckpoint` to run the Browser eventloop until the next checkpoint.  For example:

```
b.arm_timeout()
while not "foo" in b.text("#bla"):
  b.wait_checkpoint()
b.disarm_timeout()
```

The calls to `arm_timeout` and `disarm_timeout` set things up so that `b.wait_checkpoint` will throw an error eventually.